kpm/lib/kpm/diagnostic_file.rb

```diff
@@ -0,0 +1,188 @@
+
+            zipFile.add(TENANT_FILE,  tenant_export_file)
+            zipFile.add(SYSTEM_FILE,  system_export_file)
+            zipFile.add(ACCOUNT_FILE, account_export_file) unless account_id.nil?
+            zipFile.add(ZIP_LOG_FILE, log_files)
+
+          end
+
+          @logger.info "\e[32mDiagnostic data exported under #{zip_file_name} \e[0m"
+
+        else
+          raise Interrupt, 'Account id or configuration file not found'
+        end
+
+      end
+
+      # Private methods
+
+      private
+
+      def get_tenant_config
+
+        @logger.info 'Retrieving tenant configuration'
+        # this suppress the message of where it put the account file, this is to avoid confusion
+        @logger.level = Logger::WARN
+
+        @killbill_api_credentials ||= [get_config('killbill', 'api_key'), get_config('killbill','api_secret')] unless @config_file.nil?
+        @killbill_credentials ||= [get_config('killbill', 'user'), get_config('killbill','password')] unless @config_file.nil?
+        @killbill_url ||= 'http://' + get_config('killbill', 'host').to_s + ':' + get_config('killbill','port').to_s unless @config_file.nil?
+
+        tenant_config = KPM::TenantConfig.new(@killbill_api_credentials,
+                                              @killbill_credentials, @killbill_url, @logger)
+        export_file = tenant_config.export
+
+        final = TMP_DIR + File::Separator + TENANT_FILE
+        FileUtils.move(export_file, final)
+        @logger.level = @original_logger_level
+
+        final
+      end
+
+      def get_system_config
+
+        @logger.info 'Retrieving system configuration'
+        system = KPM::System.new
+        export_data = system.information(nil, true, @config_file, @kaui_web_path, @killbill_web_path)
+
+        get_system_catalina_base(export_data)
+
+        export_file = TMP_DIR + File::SEPARATOR + SYSTEM_FILE
+        File.open(export_file, 'w') { |io| io.puts export_data }
+        export_file
+      end
+
+
+      def get_account_data(account_id)
+
+        @logger.info 'Retrieving account data for id: ' + account_id
+        # this suppress the message of where it put the account file, this is to avoid confusion
+        @logger.level = Logger::WARN
+
+        account = KPM::Account.new(@config_file, @killbill_api_credentials, @killbill_credentials,
+                                   @killbill_url, @database_name,
+                                   @database_credentials,@database_host,nil, @logger)
+        export_file = account.export_data(account_id)
+
+        final  = TMP_DIR + File::Separator + ACCOUNT_FILE
+        FileUtils.move(export_file, final)
+        @logger.level = @original_logger_level
+        final
+      end
+
+      def get_log_files(log_dir)
+
+        @logger.info 'Collecting log files'
+        log_base = log_dir || @catalina_base
+        log_items = Dir.glob(log_base + File::Separator + 'logs' + File::Separator + '*')
```

Comment:
Let's assume that if the user specified a `log_path`, it is the full path and let's not append `logs` (to support things like `/var/log`).